app/templating/view_context.py

```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     return context
 
 
-def build_view_context_for_summary(metadata, answer_store, schema_context, variables, csrf_token):
-    rendered_schema_json = renderer.render(g.schema.json, **schema_context)
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block_type, variables, csrf_token,
+                                   current_location):
+
+    summary_rendered_context = None
+
+    if block_type == 'Summary':
+        sections = renderer.render(g.schema.json, **schema_context)['sections']
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block_type == 'SectionSummary':
+        for section in g.schema.json['sections']:
+            group_ids = {group['id'] for group in section['groups']}
+            if current_location.group_id in group_ids:
+                sections = [section]
```

Comment:
I thought you rendered this?

Reply:
I pass the section I'm on to the 'build_summary_rendering_context' which I thought builds and renders the page? So I just left it as that?

Reply:
so why did you call `renderer.render` for `Summary` if you thought that?

Reply:
If I make it so instead it is `g.schema.json['sections']` for `Summary` it should work?